TS项目生成缺少配置文件

生成 TypeScript 模板时，缺少 tsconfig.json 文件和 tslint.json 文件，另外依赖中 eslint 的相关插件 应该都替换成 tslint 的吧

安装并 lint 了，可以使用，但有一些问题:

+ **页面中，会出现 config 未定义**

```
  config = {
    navigationBarTitleText: '首页'
  }
```

+ **TSX 中的 JSX 不支持 `JSX not allowed in files with extension '.tsx`**

+ **组件中定义的接口类型，props 提示未定义**

+ **`defineConstants` 中定义的全局变量，在使用时会报 未定义**

以下是详细输出

```
➜  yarn run lint
yarn run v1.6.0
$ eslint src/ --ext .tsx
The react/jsx-space-before-closing rule is deprecated. Please use the react/jsx-tag-spacing rule with the "beforeSelfClo
sing" option instead.
{ type: 'MemberExpression',
  range: [ 1759, 1774 ],
  loc:
   { start: { line: 59, column: 64 },
     end: { line: 59, column: 79 } },
  object:
   { type: 'MemberExpression',
     range: [ 1759, 1769 ],
     loc: { start: [Object], end: [Object] },
     object: { type: 'ThisExpression', range: [Array], loc: [Object] },
     property:
      { type: 'Identifier',
        range: [Array],
        loc: [Object],
        name: 'state' },
     computed: false },
  property:
   { type: 'Identifier',
     range: [ 1770, 1774 ],
     loc: { start: [Object], end: [Object] },
     name: 'logo' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 2401, 2416 ],
  loc:
   { start: { line: 70, column: 33 },
     end: { line: 70, column: 48 } },
  object:
   { type: 'MemberExpression',
     range: [ 2401, 2411 ],
     loc: { start: [Object], end: [Object] },
     object: { type: 'ThisExpression', range: [Array], loc: [Object] },
     property:
      { type: 'Identifier',
        range: [Array],
        loc: [Object],
        name: 'state' },
     computed: false },
  property:
   { type: 'Identifier',
     range: [ 2412, 2416 ],
     loc: { start: [Object], end: [Object] },
     name: 'imgs' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 843, 859 ],
  loc:
   { start: { line: 36, column: 55 },
     end: { line: 36, column: 71 } },
  object:
   { type: 'ThisExpression',
     range: [ 843, 847 ],
     loc: { start: [Object], end: [Object] } },
  property:
   { type: 'Identifier',
     range: [ 848, 859 ],
     loc: { start: [Object], end: [Object] },
     name: 'handleClick' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 892, 900 ],
  loc:
   { start: { line: 36, column: 104 },
     end: { line: 36, column: 112 } },
  object:
   { type: 'Identifier',
     range: [ 892, 896 ],
     loc: { start: [Object], end: [Object] },
     name: 'item' },
  property:
   { type: 'Identifier',
     range: [ 897, 900 ],
     loc: { start: [Object], end: [Object] },
     name: 'img' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 925, 932 ],
  loc:
   { start: { line: 36, column: 137 },
     end: { line: 36, column: 144 } },
  object:
   { type: 'Identifier',
     range: [ 925, 928 ],
     loc: { start: [Object], end: [Object] },
     name: 'img' },
  property:
   { type: 'Identifier',
     range: [ 929, 932 ],
     loc: { start: [Object], end: [Object] },
     name: 'img' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 939, 947 ],
  loc:
   { start: { line: 35, column: 44 },
     end: { line: 35, column: 52 } },
  object:
   { type: 'Identifier',
     range: [ 939, 943 ],
     loc: { start: [Object], end: [Object] },
     name: 'item' },
  property:
   { type: 'Identifier',
     range: [ 944, 947 ],
     loc: { start: [Object], end: [Object] },
     name: 'url' },
  computed: false }
{ type: 'MemberExpression',
  range: [ 1049, 1057 ],
  loc:
   { start: { line: 36, column: 99 },
     end: { line: 36, column: 107 } },
  object:
   { type: 'Identifier',
     range: [ 1049, 1053 ],
     loc: { start: [Object], end: [Object] },
     name: 'item' },
  property:
   { type: 'Identifier',
     range: [ 1054, 1057 ],
     loc: { start: [Object], end: [Object] },
     name: 'img' },
  computed: false }

/Users/zacksleo/projects/mina/src/app.tsx
   7:3  error  'config' is not defined                         no-undef
  29:7  error  JSX not allowed in files with extension '.tsx'  react/jsx-filename-extension

/Users/zacksleo/projects/mina/src/components/about/index.tsx
   8:11  error  'IProps' is not defined                         no-undef
   9:5   error  'title' is not defined                          no-undef
  28:12  error  'defaultProps' is not defined                   no-undef
  35:22  error  'API_GATEWAY' is not defined                    no-undef
  49:13  error  JSX not allowed in files with extension '.tsx'  react/jsx-filename-extension
  51:36  error  Unexpected usage of doublequote                 jsx-quotes
  59:60  error  JSX 事件名需以 `on` 命名                               taro/jsx-handler-names
  70:28  error  JSX 事件名需以 `on` 命名                               taro/jsx-handler-names

✖ 26 problems (26 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

--- a/packages/eslint-plugin-taro/__tests__/jsx-handler-names.js
+++ b/packages/eslint-plugin-taro/__tests__/jsx-handler-names.js
@@ -67,6 +67,9 @@ ruleTester.run('jsx-handler-names', rule, {
     },
     {
       code: testComponent(`<View src={a.b} />`)
+    },
+    {
+      code: testComponent(`<View src={this.state.img} />`)
     }
   ],
   invalid: [{

--- a/packages/eslint-plugin-taro/rules/jsx-handler-names.js
+++ b/packages/eslint-plugin-taro/rules/jsx-handler-names.js
@@ -32,7 +32,7 @@ module.exports = {
         const propIsEventHandler = PROP_EVENT_HANDLER_REGEX.test(propKey)
         console.log(node.value.expression)
         const isPropValueFunction = source.startsWith('this.props.on') ||
-          source.startsWith('this.')
+          (source.startsWith('this.') && !source.startsWith('this.state.'))
 
         if (!propIsEventHandler && isPropValueFunction) {
           context.report({

